contracts/most/Utmost.sol

```diff
@@ -0,0 +1,150 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "../proxies/MasterCopyNonUpgradable.sol";
+import "./GenesisUtmost.sol";
+import "../utilityToken/UtilityToken.sol";
+
+/**
+ * @title Utmost contract implements UtilityToken.
+ *
+ * @dev Utmost functions as the base coin to pay for gas consumption on the
+ *      auxiliary chain.
+ */
+contract Utmost is MasterCopyNonUpgradable, GenesisUtmost, UtilityToken {
+
+    /* Events */
+
+    /** Emitted whenever the ERC20 Utmost is converted to base coin Utmost. */
+    event TokenUnwrapped(
+        address indexed _account,
+        uint256 _amount
+    );
+
+    /** Emitted whenever the base coin Utmost is converted to ERC20 Utmost. */
+    event TokenWrapped(
```

Comment:
Do we need this event as of now?

Reply:
- We already had lot of discussions in past on adding these events at OSTPrime implementation.

- `wrap` and `unwrap` are `special` Utmost contract operations(different from transfer) for base coin transactions. 

It does make sense to add these events.

Reply:
Gone through the models and transaction handlers. 
https://hackmd.io/CkFgqDFORvac2AiPFTH0Yw#ModelsRepositories
Doesn't look like we will need these events for now.